ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java

```diff
@@ -196,7 +196,8 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
     }
 
     // TODO: post-merge cleanup
-    if (!mergeCoordinator.latestValidAncestorDescendsFromTerminal(newBlockHeader)) {
+    if (!mergeContext.get().isNearHeadCheckpointSync()
```

Comment:
unit test?

Reply:
what is the behavior when we are not using checkpoint sync ? can be strange to have a method linked to the checkpoint sync in a general class. (only my opinion)

Reply:
Default of any other sync mode is going to have the check isNearHeadCheckpointSync returning false, to prevent it from being a blocker of this if.

Reply:
Also this is a very temporary piece of code that should be removed as soon as #4703 gets merged.